hdf5.spec

```diff
@@ -2,7 +2,8 @@
 %{!?_fmoddir:%global _fmoddir %{_libdir}/gfortran/modules}
 
 %global daos_major 0
-
+%global hdf5_commit 5b5a1a81029eb7bdc975beff5f18c9c440f5de56
+%define hdf5_sha .g5b5a1a8
```

Comment:
There is code in other specfiles to deduce this short value from a full commit hash.  We should leverage that so as to not have to update both above values every time.

That said, I think I am more a fan of this being a `--define`d value that the caller sets (or doesn't) so that this file is not edited for every update in addition to the `Makefile`.  DRY.

Reply:
I will move this back to the Makefile so that it is only defined once...